### PR TITLE
Optionals for AddItemCommand

### LINKED ISSUE #19: Make quantity, description and location inputs for items optional, set default values

### LINKED ISSUE #41: Test Coverage - UnitTesting Methods Rely On AB3 Depreciated Code
Related to #20 But more specific to Unittesting.
See also #40
From commit 9b2ccce:

In process of Unittesting, dificulty was encountered from utilising
testing utilities of original code base.

Specifically the assertCommandSuccess testutil method, relies on the .equals
of model manager, and model manager equality has null values initialised for AB3
and Inventoryinator versions, which if implemented to check, cause testcase
failures in stubs.

Recommended to perform refactor to code base and remove ab3 artifacts, for more effective
testing.

Will open new issue based on this finding.

_Originally posted by @blackonyyx in https://github.com/AY2021S1-CS2103T-F13-1/tp/pull/40#issuecomment-703732276_

### PR DESCRIPTION
**Pull request by blackonyyx**
resolves #41 
partially addresses #19 
Location input not fully resolved.
quantity, descriptions have optional item inputs.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddItemCommand.java [hunk 1]
```diff
@@ -33,6 +35,9 @@
 
     public static final String MESSAGE_SUCCESS = "New item added: %1$s";
     public static final String MESSAGE_DUPLICATE_ITEM = "This item already exists in the item list";
+    public static final String MESSAGE_WARNING = "New item added: %1$s \nDo note that one or more non-compulsory fields"
```
Comment (aqcd) [inline src/main/java/seedu/address/logic/commands/AddItemCommand.java]: One or more... will suffice

File: src/main/java/seedu/address/logic/parser/ItemParserUtil.java [hunk 2]
```diff
@@ -17,7 +17,11 @@
 public class ItemParserUtil {
 
     public static final String MESSAGE_INVALID_INDEX = "Index is not a non-zero unsigned integer.";
-
+    public static final String DEFAULT_QUANTITY = "0";
+    public static final String DEFAULT_LOCATION = "No location given";
+    public static final String DEFAULT_TAG = "No tag given";
+    public static final String DEFAULT_DESCRIPTION = "No description given";
```
Comment (aqcd) [inline src/main/java/seedu/address/logic/parser/ItemParserUtil.java]: perhaps just "None" is better?

File: src/test/java/seedu/address/logic/parser/AddItemCommandParserTest.java [hunk 3]
```diff
@@ -84,18 +109,18 @@ public void parse_compulsoryFieldMissing_failure() {
...
+                expectedMessage);*/
 
         // missing description prefix
-        assertParseFailure(parser, ITEM_NAME_DESC_BANANA
+        /*assertParseFailure(parser, ITEM_NAME_DESC_BANANA
                         + ITEM_QUANTITY_DESC_BANANA
                         + VALID_ITEM_DESCRIPTION_BANANA
                         + ITEM_LOCATION_DESC_SPINACH_GARDEN,
-                expectedMessage);
+                expectedMessage);*/
```
Comment (aqcd) [inline src/test/java/seedu/address/logic/parser/AddItemCommandParserTest.java]: change these tests to using wrong prefix
can remove the old tests once replaced

File: src/test/java/seedu/address/logic/parser/ArgumentTokenizerTest.java [hunk 4]
```diff
@@ -18,7 +18,7 @@
     public void tokenize_emptyArgsString_noValues() {
         String argsString = "  ";
         ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(argsString, pSlash);
-
+        System.out.println(argMultimap);
```
Comment (aqcd) [inline src/test/java/seedu/address/logic/parser/ArgumentTokenizerTest.java]: remove, same for the rest of the printlns in this file

**Review by aqcd — COMMENTED** [review]
lgtm once cosmetic fixes applied

File: docs/UserGuide.md [hunk 5]
```diff
@@ -198,6 +225,7 @@ NOT Supported as of v1.1 yet
 **EXAMPLE:**
 - `delr` <u>Bob’s 28th finger -r 1</u>
 - Deletes the **first recipe** of the item <u>Bob’s 28th finger</u>
+
 ### Delete item : `del`
 NOT Supported as of v1.1 yet
```
Comment (Rahul0506) [inline docs/UserGuide.md]: missed this

File: src/main/java/seedu/address/logic/commands/AddItemCommand.java [hunk 6]
```diff
@@ -5,8 +5,10 @@
 import static seedu.address.logic.parser.CliSyntax.PREFIX_ITEM_LOCATION;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_ITEM_NAME;
 import static seedu.address.logic.parser.CliSyntax.PREFIX_ITEM_QUANTITY;
+import static seedu.address.logic.parser.ItemParserUtil.DEFAULT_QUANTITY_TYPED;
 
 import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.ItemParserUtil;
```
Comment (Rahul0506) [inline src/main/java/seedu/address/logic/commands/AddItemCommand.java]: I think you can import only DEFAULT_DESCRIPTION separately?

**Review by Rahul0506 — APPROVED** [review]
LGTM

**Pull request merged**